### PR TITLE
feat: implement `__has_declspec_attribute`

### LINKED ISSUE #419: Implement __has_declspec_attribute
This should be fairly straightforward to implement - the functionality is almost the same as for `.macro_param_has_attribute` in `Tokenizer.zig` and `Preprocessor.zig`

Additionally, `fromStringDeclspec` in `Attribute.zig` should operate on "normalized" attribute names (with leading and trailing double-underscores removed)

Tests - uncomment lines 11-17 in `test/cases/declspec.c`

### PR DESCRIPTION
**Pull request by bingcicle**
closes #419 

CI seems to be failing, not sure if it's related to [this](https://github.com/Vexu/arocc/pull/510#issuecomment-1738617220)?

### CODE REVIEW

File: src/target.zig [hunk 1]
```diff
@@ -684,6 +684,7 @@ pub fn toLLVMTriple(target: std.Target, buf: []u8) []const u8 {
         .watchos => "watchos",
         .driverkit => "driverkit",
         .shadermodel => "shadermodel",
+        .liteos => "liteos",
```
Comment (bingcicle) [inline src/target.zig:687]: I added this in order to make it compile, but https://github.com/Vexu/arocc/pull/510 fixes this.

**Review by Vexu — CHANGES_REQUESTED** [review]
`__has_declspec_attribute(<anything>)` should return `0` when `langopts.declspec_attrs == false`.

CI failure is definitely because of llvm-17, I'll try to figure it out.

**Comment by bingcicle** [comment]
@Vexu Hmm, am I misundestanding the usage of

```
//aro-args -fdeclspec
```

at the top of the file? I was under the impression that including the args switches on the options in tests

**Comment by Vexu** [comment]
No you understand correctly, I was talking about the case where `-fdeclspec` was not specified.

**Comment by bingcicle** [comment]
Ah should I add a check there as well? I'm assuming you mean `no_declspec.c`

**Comment by Vexu** [comment]
Something like that yes.

**Comment by Vexu** [comment]
Thanks!

**Pull request merged**